cpu/stm32_common/include/periph_cpu_common.h

```diff
@@ -206,6 +206,51 @@ typedef enum {
 #endif /* ndef DOXYGEN */
 #endif /* ndef CPU_FAM_STM32F1 */
 
+#if defined(CPU_FAM_STM32F2) || defined(CPU_FAM_STM32F4) || defined(CPU_FAM_STM32F7)
+
+/**
+ * @brief   DMA configuration
+ */
+typedef struct {
+    int stream;            /**< DMA stream */
```

Comment:
What about having 2 fields: one for stream and one for channel. This would simplify the board configuration I think and only one field has to be added to UART/SPI devices. What do you think ?

Reply:
On F2/F4/F7, there is 2 DMAs, each with 8 streams. Each stream have channels. A stream can be used by only one channel at a time. And each peripheral is associated to a given stream/channel (see reference manual for stream/channel). That's why streams are declared in the config, that's the entity which needs to be locked. Declaring the stream/channel couple would make the DMA code a lot more complex because we need a lock per stream (and an interrupt per stream).